.github/workflows/pr-build-live-branch.yml

```diff
@@ -0,0 +1,125 @@
+name: Build live branch for PR
+
+on:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+env:
+  # Only need to retain for a day since the Jetpack beta builder slurps it up to distribute.
+  WCPAY_LIVE_BRANCH_ARTIFACT_RETENTION_DAYS: 1
+jobs:
+  build-and-upload-zip-file:
+    name: "Build and inform the zip file"
+    runs-on: ubuntu-20.04
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v3
+
+      - name: "Get current version"
+        id: current-version
+        run: |
+          VERSION=$(jq '.version' package.json -r)
+          echo "Current version found: $VERSION" >> $GITHUB_STEP_SUMMARY
+          echo "VERSION=$VERSION" >> $GITHUB_OUTPUT
+
+      - name: "Set up repository"
+        uses: ./.github/actions/setup-repo
+
+      - name: "Calculate the next version"
+        id: next-version
+        env:
+          RELEASE_VERSION: ${{ steps.current-version.outputs.VERSION }}
+        run: php .github/workflows/scripts/get-next-version.php # outputs.NEXT_RELEASE_VERSION
+
+      - name: "Bump dev version"
+        id: bump-dev-version
+        env: 
+          NEXT_VERSION: ${{ steps.next-version.outputs.NEXT_RELEASE_VERSION }}
+          PR_NUMBER: ${{ github.event.number }}
+          PR_HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+        run: |
+          CURRENT_VERSION=$(jq '.version' package.json -r)
+          DEV_SUFFIX_VERSION="dev_PR-${PR_NUMBER}_commit-${PR_HEAD_SHA}"
+          DEV_VERSION="${NEXT_VERSION}-${DEV_SUFFIX_VERSION}"
+          echo "Dev Version: $DEV_VERSION" >> $GITHUB_STEP_SUMMARY
+
+          # Plugin data is passed as a JSON object.
+          PLUGIN_DATA="{}"          
+          PLUGIN_DATA=$( jq -c --arg slug "woocommerce-payments" --arg ver "$DEV_VERSION" '.[ $slug ] = { version: $ver }' <<<"$PLUGIN_DATA" )
+          echo "plugin-data=$PLUGIN_DATA" >> $GITHUB_OUTPUT
+
+          # 'Version' header in woocommerce-payments.php
+          sed -i "s/^ \* Version: .*$/ * Version: $DEV_VERSION/" woocommerce-payments.php
+
+          # 'version' field in package.json
+          sed -ri 's/("version": )".*"/\1"'${DEV_VERSION}'"/' package.json
+
+          # 'Stable tag' header in readme.txt;
+          sed -i "s/^Stable tag: .*$/Stable tag: $DEV_VERSION/" readme.txt
+
+      - name: "Build the plugin"
+        id: build_plugin
+        uses: ./.github/actions/build
+
+      - name: "Update the structure of artifacts for Jetpack Beta Builder" 
+        run: |
+          cd release
+          mv woocommerce-payments woocommerce-payments-dev
+          zip -q -r "woocommerce-payments-dev.zip" "woocommerce-payments-dev/"
```

Comment:
I see the point of the double-zip, however, it has been done with the purpose to making the artifact compatible with the betadownload.jetpack.me requirements. You can see more in my PR description. That's also the reason I did not use https://github.com/Automattic/woocommerce-payments/blob/develop/.github/workflows/build-zip-and-run-smoke-tests.yml

Reply:
Okay, I see. So I'm not sure we need to point to the artifact file, or at least not without a notice to let people know they'd have to use the zip inside if they want to install it manually

Reply:
Anyway, the main purpose of this artifact is to feed into betadownload.jetpack.me.

The comment is just something nicer. I am going to expose other information in the comment content. 

 Anyway, the main purpose of this artifact is to feed into betadownload.jetpack.me.

The comment is just something nicer. I've just learned that by feeding this info into betadownload.jetpack.me, we do not need to do anything else, just use a link like this: 

- https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments={{github.head_ref}}

For example, with this PR, this link would work for any A12s (that has the access to have more options in JN) https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/5624-live-branch-support

I will update this comment to expose this info rather than show the GitHub action run link.

Reply:
Very nice.

So we wouldn't need the Tampermonkey script?

Reply:
> So we wouldn't need the Tampermonkey script?

That's right. We do not need it for a quick test. But we can add more options and it brings a better experience with the Tampermonkey script, such as the following one from Woo core. But again, this is for the other task :D 

![Markup on 2023-03-16 at 17:52:58](https://user-images.githubusercontent.com/10045087/225595800-c3bf23a3-69c4-4dc0-936c-0a0fd0d8d068.png)

Reply:
Updated the comment in this comment 4cc91b57177476138d276d7d0dfbc2806c297f59
You can also the updated comment https://github.com/Automattic/woocommerce-payments/pull/5768#issuecomment-1469972956